substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/elf/dwarf/DwarfDebugInfo.java

```diff
@@ -753,4 +777,77 @@ public int getAbstractInlineMethodIndex(ClassEntry classEntry, String methodName
         assert abstractInlineMethodIndex.get(methodName) != null : classEntry.getTypeName() + methodName;
         return abstractInlineMethodIndex.get(methodName);
     }
+
+    /**
+     * A class used to associate properties with a specific param or local whether top level or
+     * inline.
+     */
+
+    static final class DwarfLocalProperties {
```

Comment:
Would it make sense to simply add an `int index` field in `NativeImageDebugLocalValueInfo` instead of managing those indexes externally?